lib/algoliasearch-rails.rb

```diff
@@ -555,18 +555,15 @@ def algolia_reindex(batch_size = AlgoliaSearch::IndexSettings::DEFAULT_BATCH_SIZ
         master_settings.delete 'replicas'
 
         # init temporary index
-        index_name = algolia_index_name(options)
-        tmp_options = options.merge({ :index_name => "#{index_name}.tmp" })
-        tmp_options.delete(:per_environment) # already included in the temporary index_name
-        tmp_settings = settings.dup
-        tmp_index = algolia_ensure_init(tmp_options, tmp_settings, master_settings)
+        ::Algolia::copy_index!(index_name, "#{index_name}.tmp", %w(settings synonyms rules))
```

Comment:
index_name would only work for the Main index, not the indices defined via `add_index`

Reply:
Also `copy_index!` will wait forever if the src index doesnt exist, it's best to use the async method